chat_shell/chat_shell/services/streaming/core.py

```
@@ -154,20 +188,132 @@ def _slim_thinking_data(self, thinking: list) -> list:
+        """Mark current text block as complete and reset block_offset."""
+        if self.current_text_block_id:
+            self.update_block(self.current_text_block_id, {"status": "done"})
+            self.current_text_block_id = None
+            self.block_offset = 0  # Reset block offset when finalizing
+
+    def append_tool_block(
+        self,
+        tool_use_id: str,
+        tool_name: str,
+        tool_input: dict,
+        display_name: str = None,
```

Ruff (0.14.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)